src/identifier.js

```diff
@@ -47,9 +47,9 @@ Identifier.prototype.lookup_value = function (parent) {
   case '$data': return $data;
   default:
   }
-  if (token in $data) { return $data[token]; }
-  if (token in $context) { return $context[token]; }
-  if (token in globals) { return globals[token]; }
+  if (typeof $data == "object" && token in $data) { return $data[token]; }
```

Comment:
This is happening in case of foreach loop. `foreach` in case of array $data is becoming a string when an array of strings are passed and `in` operator works for only objects. It was causing failure few test cases.